src/app/views/query/query.component.js

```diff
@@ -32,25 +23,37 @@ class QueryController {
       'event.decimalLongitude',
     );
 
-    if (this.currentProject) {
-      this.handleProjectChange(this.currentProject);
-    }
-
     this.showSidebar = true;
     this.showMap = true;
     this.loading = false;
     this.sidebarToggleToolTip = 'hide sidebar';
   }
 
-  handleNewResults(results) {
+  handleNewResults(results, isAdvancedSearch) {
+    // when switching between simple and advanced searches
+    if (!results) {
+      this.revert = true; // clears table data
+      this.toggleMap(true); 
+     // return;
+    } else {
+    this.revert = false; // allows table data to be displayed
     this.results = results;
+    this.hasCoordinates = this.results.data.some(
+      d => d.event.decimalLatitude || d.event.decimalLongitude,
+    );
+    // return table view for advanced queries and queries with no lat lng data, return map view for simple queries
+    if (!this.hasCoordinates || isAdvancedSearch) {
+      this.toggleMap(false);
+    } else {
+      this.toggleMap(true);
+      }
+    }
```

Comment:
this can be simplified to `this.toggleMap(this.hasCoordinates || !isAdvancedSearch)`

Reply:
oops. It would be `this.toggleMap(hasCoordinates && !isAdvancedSearch)`